docs/modules/ROOT/partials/getting-started/con-registry-serdes-types.adoc

```diff
@@ -0,0 +1,224 @@
+    return consumer;
+}
+----
+
+
+== Using Avro SerDe with {registry}
+
+{registry} provides serializer and deserializer classes for Apache Avro out of the box, to make using Avro as
+easy as possible.  These classes are:
+
+* `io.apicurio.registry.utils.serde.AvroKafkaSerializer`
+* `io.apicurio.registry.utils.serde.AvroKafkaDeserializer`
+
+=== Configuring the Avro serializer
+
+The Avro serializer class can be configured in the following ways:
+
+* {registry} location as a URL
+* Artifact ID strategy (see the "Strategies to lookup a schema" section)
+* Global ID strategy (see the "Strategies to lookup a schema" section)
+* Global ID location
+* Global ID handler
+* Avro datum provider
+* Avro encoding
+
+==== *Global ID location*
+
+The serializer is responsible for passing the unique global ID of the schema as part of the Kafka message so that
+consumers can use the right schema for deserialization.  The location of that global ID can be in the payload of
+the message or in the message headers.  The default approach is to pass the global ID in the message payload.  If
+you want the ID sent in the message headers instead, you can set the following configuration property:
+
+`props.putIfAbsent(AbstractKafkaSerDe.USE_HEADERS, "true")`
+
+The property name is `apicurio.registry.use.headers`.
+
+
+==== *Global ID handler*
+
+When passing the global ID in the message payload, it is possible to customize precisely how that is done.  Set
+the configuration property `apicurio.registry.id-handler` to be a class that implements the
+`io.apicurio.registry.utils.serde.strategy.IdHandler` interface.  {registry} provides two implementations of
+that interface:
+
+* `io.apicurio.registry.utils.serde.strategy.DefaultIdHandler` - stores the ID as an 8 byte long
+* `io.apicurio.registry.utils.serde.strategy.Legacy4ByteIdHandler` - stores the ID as an 4 byte int
+
+{registry} represents the global ID of an artifact as a long, but for legacy reasons (or for compatibility with
+other registries or serde classes) you may want to use 4 bytes when sending the ID.
+
+==== *Avro datum provider*
+TBD
+
+==== *Avro encoding*
+
+When using Apache Avro to serializer data, it is common to use the Avro binary encoding format.  This is so that
+the data is encoded in as efficient a format as possible.  However, Avro also supports encoding the data as JSON.
+Encoding as JSON is useful because it is much easier to inspect the payload of each message, often for logging,
+debugging, or other similar use-cases.  The {registry} Avro serializer can be configured to change the encoding
+to JSON from the default (binary).
+
+Set the Avro encoding to use by configuring the `apicurio.avro.encoding` property. The value must be either
+`JSON` or `BINARY`.
+
+
+=== Configuring the Avro deserializer
+
+The Avro deserializer class must be configured to match the settings of the serializer.  As a result, the
+deserializer can be configured in the following ways:
```

Comment:
Suggest "You must configure the Avro deserializer class to match the settings of the serializer. As a result, you can configure the deserializer in the following ways:" (for active voice)